ocaml/xapi/xapi_vdi.ml

```diff
@@ -517,7 +519,8 @@ let destroy ~__context ~self =
       List.iter (fun vbd ->
           Helpers.log_exn_continue (Printf.sprintf "destroying VBD: %s" (Ref.string_of vbd))
             (fun vbd -> Db.VBD.destroy ~__context ~self:vbd) vbd) vbds;
-      (* Db.VDI.destroy ~__context ~self *)
+      (* If VDI destroyed is suspend VDI of VM then set the suspend_VDI field as null ref *)
+      List.iter (fun self -> Db.VM.set_suspend_VDI ~__context ~self ~value:Ref.null) suspend_vdi_vms;
```

Comment:
I think that one side of a relation must be a `set ref` (there are no one-to-one relations). So to do this, we'd really need to change the type of `VM.suspend_VDI` to be a `VDI set ref`, where `[]` means the same as `OpaqueRef:NULL` does now. However, we can't change the type of an existing field...